instructions.ts

```diff
@@ -67,6 +67,44 @@ export function action<T>(
   };
 }
 
+export function go<T>(
+  operation: () => Operation<T>,
+): Operation<Task<Result<T>>> {
+  function Go(frame: Frame) {
+    return shift<Result<Task<Result<T>>>>(function* (k) {
+      let child = frame.createChild();
+      let block = child.run(function* () {
+        try {
+          let value = yield* operation();
+          let result: Result<T> = {
```

Comment:
It's pretty annoying that I had to do this instead of just using the `Result` from `createFrameTask`.  I tried to get both use cases to work but ultimately it was too awkward so I opted for this solution.